internal/services/network/local_network_gateway_resource.go

```diff
@@ -139,6 +139,15 @@ func resourceLocalNetworkGatewayCreateUpdate(d *pluginsdk.ResourceData, meta int
 	// There is a bug in the provider where the address space ordering doesn't change as expected.
 	// In the UI we have to remove the current list of addresses in the address space and re-add them in the new order and we'll copy that here.
 	if !d.IsNewResource() && d.HasChange("address_space") {
+
+		// since the local network gateway cannot have both empty address prefix and empty BGP setting(confirmed with service team, it is by design),
+		// replace the empty address prefix with the first address prefix in the "address_space" list to avoid error.
+		if v := d.Get("address_space").([]interface{}); len(v) > 0 {
```

Comment:
what happens if `address_space` is removed during an update? wouldn't this error still occur?

Reply:
If `address_space` is removed during an update, I think the error "The local network gateway /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/rg-example/providers/Microsoft.Network/localNetworkGateways/example cannot have both empty address prefix and empty BGP Settings." is expected since this is by design. It's just that this error is returned from the API but not intercepted by Terraform Provider. Do you mean "address prefix and BGP Settings cannot be empty at the same time" should be appended in Terraform Provider but not returned by the API?

Reply:
Any updates?